tests/cache/tests.py

```diff
@@ -440,15 +440,15 @@ def test_expiration(self):
     def test_touch(self):
         # cache.touch() updates the timeout.
         cache.set('expire1', 'very quickly', timeout=1)
-        self.assertIs(cache.touch('expire1', timeout=4), True)
-        time.sleep(2)
+        self.assertIs(cache.touch('expire1', timeout=2), True)
+        time.sleep(1.0)
         self.assertIs(cache.has_key('expire1'), True)
-        time.sleep(3)
+        time.sleep(1.5)
         self.assertIs(cache.has_key('expire1'), False)
         # cache.touch() works without the timeout argument.
         cache.set('expire1', 'very quickly', timeout=1)
         self.assertIs(cache.touch('expire1'), True)
-        time.sleep(2)
+        time.sleep(1.5)
```

Comment:
I'm not sure about these changes, `test_touch` fails quite often even with longer delays.

Reply:
The amount of time wasted sleeping in these tests is multiplied up for the 8 cache backends making it rather significant as noted in the commit message.

I understand that concern as I've noticed that it does fail occasionally. In my testing locally, however, and in the number of times that I've rebased this and had the tests running on Jenkins, I've not seen any noticeable increase in the rate of failure.

I guess it's hard to prove that it isn't worse in a relatively limited number of runs. Perhaps the only way is to take the change and revert it if it is noticeably worse after a few days or weeks. My impression has been that when it fails, it would fail even if you gave it 5 seconds here.

It's hard because it is just such a "nasty" test and there is no other way to make this faster due to the whole second timeouts in memcached.

(Although we could also possibly use shorter timeouts if it is a non-memached backend...)